advcpmv-0.8-8.32.patch

```diff
@@ -549,77 +674,105 @@ diff -aur coreutils-8.32/src/cp.c coreutils-8.32-patched/src/cp.c
          case 'H':
            x.dereference = DEREF_COMMAND_LINE_ARGUMENTS;
            break;
-diff -aur coreutils-8.32/src/mv.c coreutils-8.32-patched/src/mv.c
+diff -aur coreutils-8.32/src/mv.c coreutils-8.32-patched-nc/src/mv.c
 --- coreutils-8.32/src/mv.c	2020-01-01 15:13:12.000000000 +0100
-+++ coreutils-8.32-patched/src/mv.c	2021-08-11 19:40:37.729820358 +0200
++++ coreutils-8.32-patched-nc/src/mv.c	2022-02-13 20:39:10.620070315 +0100
 @@ -66,6 +66,7 @@
    {"target-directory", required_argument, NULL, 't'},
    {"update", no_argument, NULL, 'u'},
    {"verbose", no_argument, NULL, 'v'},
-+  {"progress-ar", no_argument, NULL, 'g'},
++  {"progress-bar", no_argument, NULL, 'g'},
    {GETOPT_HELP_OPTION_DECL},
    {GETOPT_VERSION_OPTION_DECL},
    {NULL, 0, NULL, 0}
-@@ -168,10 +169,86 @@
- static bool
- do_move (const char *source, const char *dest, const struct cp_options *x)
+@@ -170,8 +171,130 @@
  {
-+  struct timeval start_time;
-+
    bool copy_into_self;
    bool rename_succeeded;
++
++  /* BEGIN progress mod */
++  struct timeval start_time;
++
 +  if(progress && x->rename_errno != 0) {
-+    /* BEGIN progress mod */
-+    g_iTotalSize = 0;
-+    g_iFilesCopied = 0;
-+    g_iTotalWritten = 0;
++    if (g_iTotalSize == 0)
++      g_iTotalSize = 0;
++    if (g_iTotalFiles == 0)
++      g_iTotalFiles = 0;
++    if (g_iFilesCopied == 0)
++      g_iFilesCopied = 0;
++    if (g_iDirectoriesCopied == 0)
++      g_iDirectoriesCopied = 0;
++    if (g_iTotalWritten == 0)
++      g_iTotalWritten = 0;
++
++    if (target_directory_operand (source))
++      g_iDirectoriesCopied++;
 +
 +    gettimeofday (& start_time, NULL);
 +    g_oStartTime = start_time;
 +
-+    printf ("Calculating total size... \r");
++    /* how many files are we copying */
++    FILE *fp ;
++    char output[1024];
++    char fcmd[] = "find";
++    fp = spawn(fcmd, (char *[]){ fcmd, (unsigned char *)(size_t)source, NULL, "-type", "f", NULL });
```

Comment:
Same here, remove the NULL from the middle.

Also this looks insanely sus, `(unsigned char *)(size_t)source` ; what exactly is `source` defined as?

Reply:
```
do_move (char const *source, char const *dest, const struct cp_options *x)
```

Reply:
Ahh, so its `const` causing the problem. Yeah just casting to `(char *)` to get rid of the const is fine.

Also just FYI the reason why `spawn()` isnt taking a const char * is because execvp() doesn't have const in its parameter either. `man 3p exec` says that the functions shouldn't modify it's parameters, but for some reason it's not declared as const. /shrug

Reply:
Thank you very much for the explanation. :1st_place_medal: I'll supply the modifications soon.